### PR TITLE
Clean up old entries in etcd on request only.

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -36,7 +36,8 @@ CMD cd /mnt/ctlog/ && \
         --trusted_cert_file=/usr/local/etc/ctlog_ca_roots.pem \
         --stderrthreshold=0 \
         --log_dir=/mnt/ctlog/logs \
-        --tree_signing_frequency_seconds=60 \
+        --tree_signing_frequency_seconds=30 \
+        --guard_window_seconds=10 \
```
Comment (pphaneuf) [inline Dockerfile]: That's for testing only, no? Sure you want to push this, especially in this PR (that looks like this hasn't got much to do with it)?

Comment (AlCutter) [inline Dockerfile]: Yeah, I'll pull it out into a separate PR though and merge it in anyway, then we can remove it once things have settled down a bit.

File: cpp/log/etcd_consistent_store-inl.h [hunk 2]
```diff
@@ -625,14 +619,18 @@ void EtcdConsistentStore<Logged>::OnEtcdServingSTHUpdated(
...
-util::Status EtcdConsistentStore<Logged>::RunOneCleanUpIteration(
-    const int64_t clean_up_to_sequence_number) {
-  CHECK_LE(0, clean_up_to_sequence_number);
+util::Status EtcdConsistentStore<Logged>::CleanupOldEntries() {
   if (!election_->IsMaster()) {
     return util::Status(util::error::PERMISSION_DENIED,
                         "Non-master node cannot run cleanups.");
   }
 
+  if (!serving_sth_) {
```
Comment (pphaneuf) [inline cpp/log/etcd_consistent_store-inl.h]: Here and at line 641, you're accessing ```serving_sth_``` without holding ```mutex_```? Isn't that wrong? Could probably deal with this by calling ```GetServingSTH```, which would give you a copy (or tell you that there's no serving STH).

Comment (AlCutter) [inline cpp/log/etcd_consistent_store-inl.h]: Grr, yep - that was a bug in the original version and I didn't notice it here.

Comment (pphaneuf) [inline cpp/log/etcd_consistent_store-inl.h]: In the original, I think this checking was done with the lock held in the ```CleanUpEntriesThread``` loop?

Comment (AlCutter) [inline cpp/log/etcd_consistent_store-inl.h]: Nope, checkout line 712 in the old version...

**Comment by pphaneuf** [comment]
LGTM, but I have a question on why splitting the sequencing and signing... This is so we put out STHs in a more timely manner? If I'm not mistaken, if there's a huge pile of new certs, this will now issue STHs while sequencing is done (so more incrementally, instead of once after everything is sequenced)?

**Comment by AlCutter** [comment]
Yeah, that's exactly it.

**Pull request merged**